### PR TITLE
[Feature] Add document font controller

### PR DESCRIPTION
**Pull request by pietervp**
This PR adds FontController to the SDK

## Related tickets

-   [WRS-777](https://support.chili-publish.com/projects/WRS/issues/WRS-777)

### CODE REVIEW

**Comment by pkgacek** [comment]
Description updated.

File: src/controllers/FontController.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+    /**
+     * @ignore
+     */
+    constructor(editorAPI: EditorAPI) {
+        this.#editorAPI = editorAPI;
+    }
+
+    /**
+     * This method returns the list of colors
```
Comment (brapoprod) [inline src/controllers/FontController.ts:23]:
```suggestion
     * This method returns the list of fonts

```

File: src/controllers/FontController.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+     * @returns
+     */
+    getFonts = async () => {
+        const res = await this.#editorAPI;
+        return res.getFonts().then((result) => getEditorResponseData<DocumentFont[]>(result));
+    };
+
+    /**
+     * This method returns a color by id
```
Comment (brapoprod) [inline src/controllers/FontController.ts:32]:
```suggestion
     * This method returns a font by id

```

File: src/controllers/FontController.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+     * This method removes a color
+     * @param id The ID of a specific color
+     * @returns
+     */
+    removeFont = async (id: Id) => {
+        const res = await this.#editorAPI;
+        return res.removeFont(id).then((result) => getEditorResponseData<null>(result));
+    };
+
+    addFont = async (connectorId: Id, font: AddDocumentFont) => {
```
Comment (brapoprod) [inline src/controllers/FontController.ts:51]: needs documentation

File: src/controllers/FontController.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    /**
+     * Check if the font is used anywhere in the document
+     * @param fontKey The ID of the font to check
+     * @returns Whether the font is used 
+     */
+    isFontUsed = async (fontKey: string) => {
+        const res = await this.#editorAPI;        
+        return res.isFontUsed(fontKey).then((result) => getEditorResponseData<boolean>(result));    
+    }
+}
```
Comment (brapoprod) [inline src/controllers/FontController.ts:65]: Change color to fonts for every method

File: types/FontConnectorTypes.ts [hunk 5]
```diff
@@ -8,8 +8,8 @@ export enum FontDownloadType {
 export type Font = {
     id: string;
     name: string;
-    fontFamily?: string | null;
-    fontStyle?: string | null;
+    family?: string | null;
+    style?: string | null;
```
Comment (brapoprod) [inline types/FontConnectorTypes.ts:12]: can it be undefined or is it always null?

**Review by brapoprod — CHANGES_REQUESTED** [review]

File: src/controllers/FontController.ts [hunk 6]
```diff
@@ -0,0 +1,65 @@
+import { EditorAPI, Id } from '../../types/CommonTypes';
+import { AddDocumentFont, DocumentFont } from '../../types/FontTypes';
+import { getEditorResponseData } from '../utils/EditorResponseData';
+
+/**
+ * The FontController is responsible for all communication regarding color styles.
```
Comment (alexandraFlavia9) [inline src/controllers/FontController.ts:6]: regarding font styles

File: types/FontConnectorTypes.ts (hunk 5 above)
Comment (pkgacek) [reply types/FontConnectorTypes.ts:12]: @pietervp

Comment (pietervp) [reply types/FontConnectorTypes.ts:12]: I think `null` would be the only supported value. This type is part of the connector contract which demands this property as a return type.

Comment (brapoprod) [reply types/FontConnectorTypes.ts:12]: then the ? can be removed

**Review by brapoprod — APPROVED** [review]

**Review by pkgacek — APPROVED** [review]

**Pull request merged**